plugins/nominal-connection-checker/test/type_hierarchy_test.mocha.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Unit tests for TypeHierarchy.
+ */
+
+const chai = require('chai');
+
+const {TypeHierarchy} = require('../src/type_hierarchy');
+
+suite('TypeHierarchy', function() {
+  suite('typeExists', function() {
+    test('Simple', function() {
+      const hierarchy = new TypeHierarchy({
+        'A': { },
+      });
+      chai.assert.isTrue(hierarchy.typeExists('A'));
+    });
+
+    test('Case', function() {
+      const hierarchy = new TypeHierarchy({
+        'A': { },
+      });
+      chai.assert.isFalse(hierarchy.typeExists('a'),
```

Comment:
What benefit do you see in case-sensitivity?  I learn toward case-insensitivity for this application, again with the goal of reducing the impact of typos.